cmk/base/plugins/agent_based/fortigate_antivirus.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# author: Oguzhan Cicek, OpenSource Security GmbH - oguzhan(at)os-s.de
+
+#Example Input:
+#.1.3.6.1.2.1.1.1.0 test
+#.1.3.6.1.2.1.1.2.0 .1.3.6.1.4.1.12356.101.1.15000
+#[...]
+#.1.3.6.1.4.1.12356.101.8.2.1.1.1.102 1
+#.1.3.6.1.4.1.12356.101.8.2.1.1.1.103 2
+#.1.3.6.1.4.1.12356.101.8.2.1.1.1.104 3
+#.1.3.6.1.4.1.12356.101.8.2.1.1.1.105 4
+#.1.3.6.1.4.1.12356.101.8.2.1.1.1.106 5
+#.1.3.6.1.4.1.12356.101.8.2.1.1.2.102 11
+#.1.3.6.1.4.1.12356.101.8.2.1.1.2.103 22
+#.1.3.6.1.4.1.12356.101.8.2.1.1.2.104 33
+#.1.3.6.1.4.1.12356.101.8.2.1.1.2.105 44
+#.1.3.6.1.4.1.12356.101.8.2.1.1.2.106 55
+#[...]
+
+#Example GUI Output:
+#OK	FortiGate AntiVirus Info: 0
+#       Average amount of Viruses in last 5 Min.: detected: 0.00, blocked: 0.00
+#OK	FortiGate AntiVirus Info: 1
+#       Average amount of Viruses in last 5 Min.: detected: 0.00, blocked: 0.00
+#OK	FortiGate AntiVirus Info: 2
+#       Average amount of Viruses in last 5 Min.: detected: 0.00, blocked: 0.00
+#OK	FortiGate AntiVirus Info: 3
+#       Average amount of Viruses in last 5 Min.: detected: 0.00, blocked: 0.00
+#OK	FortiGate AntiVirus Info: 4
+#       Average amount of Viruses in last 5 Min.: detected: 0.00, blocked: 0.00
+
+from .agent_based_api.v1.type_defs import (
+    DiscoveryResult,
+    CheckResult,
+    StringTable,
+)
+from .agent_based_api.v1 import (
+    register,
+    get_average,
+    Service,
+    startswith,
+    Metric,
+    SNMPTree,
+    check_levels,
+    get_value_store,
+)
+from typing import List, Mapping, Tuple
+import time
+
+Section = List[List[str]]
+
+
+def parse_fortigate_antivirus(string_table: List[StringTable]) -> Section:
+    return string_table[0]
+
+
+def discovery_fortigate_antivirus(section: Section) -> DiscoveryResult:
+    for pos in range(0, len(section)):
+        yield Service(item=str(pos))
+
+
+def check_fortigate_antivirus(item: str, params: Mapping[str, Tuple[float, float]],
+                              section: Section) -> CheckResult:
+    detected_levels = params.get("detections")
+    detected = int(section[int(item)][0])
+    blocked = int(section[int(item)][1])
+    value_store = get_value_store()
+    now = time.time()
+    detected_rate = get_average(value_store, ("detected_rate_%s" % (item)), now, detected, 5)
```

Comment:
This seems wrong. OID description:
fgAvVirusDetected: Number of virus transmissions detected in the virtual domain since start-up 
To me, this seems like a cumulative counter which can only increase (or reset back to zero). Please check this. If this is indeed a counter, you need to use get_rate (and the unit would then be blocks/sec).